tests/regr_test.py

```diff
@@ -0,0 +1,184 @@
+        "Note that this cannot be specified if --all is also specified."
+    ),
+)
+parser.add_argument(
+    "-p",
+    "--python-version",
+    dest="versions_to_test",
+    choices=SUPPORTED_VERSIONS,
+    nargs="*",
+    action="extend",
+    help=(
+        "Run mypy for certain Python versions (defaults to sys.version_info[:2])"
+        "Note that this cannot be specified if --all is also specified."
+    ),
+)
+
+
+def get_recursive_requirements(package_name: str, seen: set[str] | None = None) -> list[str]:
+    seen = seen if seen is not None else {package_name}
+    for dependency in filterfalse(seen.__contains__, read_dependencies(package_name)):
+        seen.update(get_recursive_requirements(dependency, seen))
+    return sorted(seen | {package_name})
+
+
+def test_testcase_directory(package: PackageInfo, version: str, platform: str) -> ReturnCode:
+    package_name, test_case_directory = package
+    IS_STDLIB = package_name == "stdlib"
+
+    msg = f"Running mypy --platform {platform} --python-version {version} on the "
+    msg += "standard library test cases..." if IS_STDLIB else f"test cases for {package_name!r}..."
+    print(msg, end=" ")
+
+    flags = [
+        "--python-version",
+        version,
+        "--show-traceback",
+        "--show-error-codes",
+        "--no-error-summary",
+        "--platform",
+        platform,
+        "--no-site-packages",
+        "--strict",
+    ]
+
+    # --warn-unused-ignores doesn't work for files inside typeshed.
+    # SO, to work around this, we copy the test_cases directory into a TemporaryDirectory.
+    with tempfile.TemporaryDirectory() as td:
+        td_path = Path(td)
+        new_test_case_dir = td_path / "test_cases"
+        shutil.copytree(test_case_directory, new_test_case_dir)
+        env_vars = dict(os.environ)
+        if IS_STDLIB:
+            flags.extend(["--custom-typeshed-dir", str(Path(__file__).parent.parent)])
+        else:
+            # HACK: we want to run these test cases in an isolated environment --
+            # we want mypy to see all stub packages listed in the "requires" field of METADATA.toml
+            # (and all stub packages required by those stub packages, etc. etc.),
+            # but none of the other stubs in typeshed.
+            #
+            # The best way of doing that without stopping --warn-unused-ignore from working
+            # seems to be to create a "new typeshed" directory in a tempdir
+            # that has only the required stubs copied over.
+            new_typeshed = td_path / "typeshed"
+            os.mkdir(new_typeshed)
+            shutil.copytree(Path("stdlib"), new_typeshed / "stdlib")
+            requirements = get_recursive_requirements(package_name)
+            for requirement in requirements:
+                shutil.copytree(Path("stubs", requirement), new_typeshed / "stubs" / requirement)
+            sep = ";" if sys.platform == "win32" else ":"
```

Comment:
This is `os.pathsep` (not to be confused with `os.path.sep`).

Reply:
Huh, TIL!